### PR TITLE
ARROW-17113: [Java] Fail loudly in static initializer blocks

### PR DESCRIPTION
**Pull request by lidavidm**
- MemoryUtil will print a stack trace to the console on failure (since Arrow won't really work without it)
- Flight utility classes will print a stack trace but won't prevent class initialization (it will just fall back to the "slow" path)

### CODE REVIEW

**Review by lwhite1 — APPROVED** [review]
LGTM

File: java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java
Comment (mystic-lama) [inline java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java:139]: Should we redirect the stacktrace to logger.error?

File: java/memory/memory-netty/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java
Comment (mystic-lama) [inline java/memory/memory-netty/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java:37]: Is this just cleanup? trying to understand it's relation with JIRA

**Review by paliwalashish — APPROVED** [review]
+1

File: java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java
Comment (lidavidm) [reply java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java:139]: I wanted this to fail as loudly as possible since Arrow just won't work without it

File: java/memory/memory-netty/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java
Comment (lidavidm) [reply java/memory/memory-netty/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java:37]: These were unused, I was examining all static initializer blocks so I just decided to remove these

File: java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java
Comment (mystic-lama) [reply java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java:139]: Most cases this won't be an issue since console output is also redirected to logs. Only a very small percentage who miss redirecting console output to logs shall have to look further.

Comment (lidavidm) [reply java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java:139]: Interesting, in my experience only explicit logs made it to the logfiles (and java.util.logging wasn't consistently redirected)…though maybe that's not a great example

Comment (mystic-lama) [reply java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java:139]: In my experience, we always had System error/out redirected to a log file and the logger log file is also there. Eventually all the logs make it into Splunk.
Anyways nothing major, we can always refine later. Let's plan to merge the PR once committers approve

**Comment by pitrou** [comment]
@lidavidm Should this be merged?

**Comment by lidavidm** [comment]
Thanks for the reminder, this slipped under my radar

**Pull request merged**
